locust/contrib/fasthttp.py

```diff
@@ -0,0 +1,383 @@
+                      SSLError, Timeout, HTTPConnectionClosed)
+
+
+def _construct_basic_auth_str(username, password):
+    """Construct Authorization header value to be used in HTTP Basic Auth"""
+    if isinstance(username, str):
+        username = username.encode('latin1')
+    if isinstance(password, str):
+        password = password.encode('latin1')
+    return 'Basic ' + b64encode(b':'.join((username, password))).strip().decode("ascii")
+
+
+class FastHttpLocust(Locust):
+    """
+    Represents an HTTP "user" which is to be hatched and attack the system that is to be load tested.
+    
+    The behaviour of this user is defined by the task_set attribute, which should point to a 
+    :py:class:`TaskSet <locust.core.TaskSet>` class.
+    
+    This class creates a *client* attribute on instantiation which is an HTTP client with support 
+    for keeping a user session between requests.
+    """
+    
+    client = None
+    """
+    Instance of HttpSession that is created upon instantiation of Locust. 
+    The client support cookies, and therefore keeps the session between HTTP requests.
+    """
+    
+    def __init__(self):
+        super(FastHttpLocust, self).__init__()
+        if self.host is None:
+            raise LocustError("You must specify the base host. Either in the host attribute in the Locust class, or on the command line using the --host option.")
+        if not re.match(r"^https?://[^/]+$", self.host, re.I):
+            raise LocustError("Invalid host (`%s`). The specified host string must be a base URL without a trailing slash. E.g. http://example.org" % self.host)
+        
+        self.client = FastHttpSession(base_url=self.host)
+
+
+class FastHttpSession(object):
+    auth_header = None
+    
+    def __init__(self, base_url):
+        self.base_url = base_url
+        self.cookiejar = CookieJar()
+        self.client = LocustUserAgent(max_retries=1, cookiejar=self.cookiejar)
+        
+        # Check for basic authentication
+        parsed_url = urlparse(self.base_url)
+        if parsed_url.username and parsed_url.password:
+            netloc = parsed_url.hostname
+            if parsed_url.port:
+                netloc += ":%d" % parsed_url.port
+            
+            # remove username and password from the base_url
+            self.base_url = urlunparse((parsed_url.scheme, netloc, parsed_url.path, parsed_url.params, parsed_url.query, parsed_url.fragment))
+            # store authentication header (we construct this by using _basic_auth_str() function from requests.auth)
+            self.auth_header = _construct_basic_auth_str(parsed_url.username, parsed_url.password)
+    
+    def _build_url(self, path):
+        """ prepend url with hostname unless it's already an absolute URL """
+        if absolute_http_url_regexp.match(path):
+            return path
+        else:
+            return "%s%s" % (self.base_url, path)
```

Comment:
this might be a bit confusing if you have a something like:
```
base_url = http://localhost.com
path = myservice
```

Reply:
Yeah, maybe. Shouldn't be too hard to figure out the exception though. We currently have the same potential problem with the default HTTP client as well.